### DISCUSSION TITLE
[Question] Different Response from /api/v1/dashboard when executing via remote vs /swagger/v1 example

### DISCUSSION THREAD

**Discussion by hitzelc** [apache/superset#15460] [Q&A / Help]
Is there any reason why a remote request to /api/v1/dashboard would return a different quantity of dashboards than a request executed locally via the /swagger/v1 page?

In the former case I get a response with count 1 (incorrect), and in the latter case I get a response with count 16 (correct).

It doesn't appear to be role or permission based, as in both cases I am logged in as admin and I am authorized with the same JWT token in both cases. I am not including any additional query parameters in either case.

I am certain I must be doing something wrong, but it is not clear to me what from the docs. Any pointers in the right direction would be greatly appreciated.

**Comment by codegeekjk** [comment]
How you are login via an API can you please help me that i.e how you have login and then calling xxxx:8088/api/v1/dashboard
can you help me with the API request

**Reply by hitzelc** [reply]
I make a POST request to /security/login per the example here https://superset.apache.org/docs/rest-api
`BASE_URL = 'http://XXXXXX:3033/api/v1'` (3033 is not a typo, that is the port I have configured).
`body = {'password':'XXXX','username':'XXXX','refresh':True,'provider':'db'}`
`r = requests.post(url=BASE_URL+'/security/login',json=body)`
`ACCESS_TOKEN = json.loads(r.text).get('access_token')`
`REFRESH_TOKEN = json.loads(r.text).get('refresh_token')`

The user I am using to log-in has Admin privileges.

I receive an access token and a refresh token.

I then make a GET request to /dashboard/ with no query parameter and the following header
`header = {'Authorization': f'Bearer {ACCESS_TOKEN}','Content-Type':'application/json'}`
`requests.get(url=f'{BASE_URL}/dashboard',headers=header)`

I have also tried excluding the header (not that that makes any sense).

I receive a response in the expected format, but this response contains a single dashboard (whereas the response data when executing the example request on the /swagger/v1 page contains sixteen dashboards). Both requests are identical to the requests on the http://XXXXXX:3033/swagger/v1 page in format near as I can tell. I also do not see anything odd going on with permissions or settings on this single dashboard (i.e. I see no reason that this single dashboard is any different from the other fifteen dashboards).

**Reply by nytai** [reply]
That's quite strange. You could try adding some request params, eg 

`/api/v1/dashboard/?q=(page:0,page_size:25)`

**Reply by hitzelc** [reply]
> That's quite strange. You could try adding some request params, eg
> 
> `/api/v1/dashboard/?q=(page:0,page_size:25)`

Thank you. I still seem to only get a single dashboard when executing this remotely. I wonder if somehow this could have something to do with something in my config or some sort of access privileges I am unaware of (though if it is the same login for authorization in both cases I am not sure this could really be the culprit)? I'll keep messing around and if I make any progress I will reply here, very strange.

**Reply by hitzelc** [reply]
No real progress on this issue unfortunately -- it seems that the user is not set correctly in Superset, as the logs under Security -> Action Log show the User as "None" for every API action. This is unexpected, since the POST to login successfully returns an access token, and at no point to I get a failure on the Authorization part of the header. I am also including the csrf token in my header, so it is unclear to me what exactly is falling down.

I searched the issues on the Superset repo pretty extensively, the only other time I've seen issues with setting the user seems to be around PUBLIC_ROLE_LIKE, AUTH_ROLE_LIKE, etc. configurations, which I do not have set in my superset_config file.

Have you ever seen something like this? A User of "None" despite valid access and csrf tokens? Any help or pointers are appreciated.

EDIT: This issue #13907 appears to be the best related issue. Same login, one set of requests done via web/swagger and the other done via REST API, different results. Seems like a dead issue, but I have to assume the same thing is going on in these cases.

**Reply by hitzelc** [reply]
I found another possibly related issue: #16398

And the workaround from that issue seems to do the trick in my case.

https://github.com/metriql/metriql-superset/blob/main/metriql2superset/superset.py#L21

So it seems that the csrf token and access token that I get via the REST API do not function as expected when executing further requests, but if I get the csrf token and access token from a post to the normal /login/ page as in the linked workaround, everything does work as expected!

**Comment by giacomochiarella** [comment]
old post same issue [here](https://github.com/apache/superset/issues/25890)
